pkg/apis/eventing/v1alpha1/trigger_validation.go

```diff
@@ -34,7 +36,13 @@ var (
 
 // Validate the Trigger.
 func (t *Trigger) Validate(ctx context.Context) *apis.FieldError {
-	return t.Spec.Validate(ctx).ViaField("spec")
+	errs := t.Spec.Validate(ctx).ViaField("spec")
+	dependencyAnnotation, ok := t.GetAnnotations()[DependencyAnnotation]
+	if ok {
+		dependencyAnnotationPrefix := "annotation." + DependencyAnnotation
```

Comment:
I suggest this prefix: `fmt.Sprintf("metadata.annotations[%s]", DependencyAnnotation)`. That's unambiguous about which key is causing the error.